components/ViewToggle.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+
+interface ViewToggleProps {
+  onClick?: (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
+    state: 'grid' | 'list',
+  ) => void;
+  value?: 'grid' | 'list';
+}
+
+const ViewToggle: React.FC<ViewToggleProps> = (props) => {
+  const { onClick, value } = props;
+
+  const [state, setState] = React.useState(value ?? 'grid');
```

Comment:
Let's make this component stateless and [_lift state up_](https://reactjs.org/docs/lifting-state-up.html) to the parent, passing only props here. This will help reusability.